sentry-uitest/sentry-uitest-android-benchmark/src/androidTest/java/io/sentry/uitest/android/benchmark/SentryBenchmarkTest.kt

```diff
@@ -0,0 +1,137 @@
+        context = ApplicationProvider.getApplicationContext()
+        context.cacheDir.deleteRecursively()
+        IdlingRegistry.getInstance().register(BenchmarkActivity.scrollingIdlingResource)
+        // Must run on the main thread to get the main thread choreographer.
+        runner.runOnMainSync {
+            choreographer = Choreographer.getInstance()
+        }
+    }
+
+    @AfterTest
+    fun cleanup() {
+        IdlingRegistry.getInstance().unregister(BenchmarkActivity.scrollingIdlingResource)
+    }
+
+    @Test
+    fun benchmarkSameOperation() {
+
+        // We compare two operation that are the same. We expect the increases to be negligible, as the results
+        // should be very similar.
+        val op1 = BenchmarkOperation(choreographer, getOperation(runner))
+        val op2 = BenchmarkOperation(choreographer, getOperation(runner))
+        val comparisonResult = BenchmarkOperation.compare(op1, "Op1", op2, "Op2")
+
+        assertTrue { comparisonResult.durationIncrease in -1F..1F }
+        assertTrue { comparisonResult.cpuTimeIncrease in -1F..1F }
+        // The fps decrease is skipped for the moment, due to approximation:
+        // if an operation runs at 59.49 fps and the other at 59.51, they are considered 59 and 60 fps respectively.
+        // Their difference would be 1 / 60 * 100 = 1.66666%
+        // On slow devices, a difference of 1 fps on an average of 20 fps would account for 5% decrease.
+        // On even slower devices the difference would be even higher. Let's skip for now, as it's not important anyway.
+//        assertTrue { comparisonResult.fpsDecrease in -2F..2F }
+        assertTrue { comparisonResult.droppedFramesIncrease in -1F..1F }
+    }
+
+    @Test
+    fun benchmarkProfiledTransaction() {
+
+        runner.runOnMainSync {
+            SentryAndroid.init(context) { options: SentryOptions ->
+                options.dsn = "https://key@uri/1234567"
+                options.isEnableAutoSessionTracking = false
+                options.tracesSampleRate = 1.0
+                options.isTraceSampling = true
+                options.isProfilingEnabled = true
+            }
+        }
+
+        // We compare the same operation with and without profiled transaction.
+        // We expect the profiled transaction operation to be slower, but not slower than 5%.
+        val benchmarkOperationNoTransaction = BenchmarkOperation(choreographer, getOperation(runner))
+        val benchmarkOperationProfiled = BenchmarkOperation(
+            choreographer,
+            getOperation(runner) {
+                Sentry.startTransaction("Benchmark", "ProfiledTransaction")
+            }
+        )
+        val comparisonResult = BenchmarkOperation.compare(
+            benchmarkOperationNoTransaction,
+            "NoTransaction",
+            benchmarkOperationProfiled,
+            "ProfiledTransaction"
+        )
+
+        assertTrue { comparisonResult.durationIncrease in -1F..5F }
```

Comment:
I think it should be in the range of 0..5, right?

Reply:
yeah, but on high end devices with several threads available, a single thread can manage profiling by itself, and while the cpu usage increases, the other metrics could remain the same.
I was thinking about just setting upper thresholds for that.
Of course, on real devices of SauceLabs, since they are slower, we expect all the metrics to increase, even if not much

Reply:
Will put back the range 0..5 👍